tests/contrib/sqlalchemy_1/sql_alchemy_plugin/test_sql_alchemy_plugin_integration.py

```diff
@@ -75,3 +78,29 @@ def test_serializing_a_list_of_sql_alchemy_instances() -> None:
         response = client.post("/companies/bulk", json=serialized_companies)
         assert response.status_code == HTTP_201_CREATED
         assert response.json() == serialized_companies
+
+
+@pytest.mark.xfail(reason="To demonstrate undesired behavior")
+def test_sqlalchemy_plugin_with_sqlalchemy_dependency() -> None:
+    from tests.contrib.sqlalchemy_1.sql_alchemy_plugin.models import SQLAlchemyBase
+
+    class Foo(SQLAlchemyBase):
+        __tablename__ = "foo"
+        id = Column(Integer, primary_key=True)
+
+    foo_instance = Foo(id=1)
+
+    def provide_foo() -> Foo:
+        return foo_instance
+
+    @get()
+    def get_handler(foo: Foo) -> Foo:
+        assert foo is foo_instance
+        return foo
+
+    with create_test_client(
+        [get_handler], plugins=[SQLAlchemyPlugin()], dependencies={"foo": Provide(provide_foo)}, debug=True
+    ) as client:
+        response = client.get("/")
+        assert response.status_code == HTTP_200_OK
+        assert response.json() == {"id": foo_instance.id}
```

Comment:
This test was added for demonstration purposes.

The test fails on `assert foo is foo_instance` in the handler, because the return value of `provide_foo` is used to populate a pydantic model that represents `Foo`, and then the values from the pydantic model are used to populate values on a new `Foo` instance that eventually gets injected.

I think this is surprising behavior for dependencies and that validation of individual attributes every time a plugin supported type is injected into some function is overkill.

Reply:
@litestar-org/maintainers

Reply:
> I think this is surprising behavior for dependencies and that validation of individual attributes every time a plugin supported type is injected into some function is overkill.

Absolutely agree. Not only is it surprising but also undesirable. Instances might be expensive to create, contain hidden state or users might simply - and rightfully so - rely on `a is b` logic and expect it to work.

Reply:
Yup 👍